### PR TITLE
chore: Pull JS Linting to separate Action

### PR DESCRIPTION
**Pull request by nschonni**
### Why:

Reduce number of jobs, so the JS linting is only run if JS files are changed. I left the duplicate job run in the test-translations.yml, but it could also probably be removed if the branch filters are updated here

### What's being changed:

A separate linting job that uses the pinned version of `standard` rather than a floating version from `npx` run only when JS files are changed. Relocated the previous `--fix` version of the script to a separate target

### Check off the following:
- [x] All of the tests are passing.
- [x] I have reviewed my changes in staging.
- [x] For content changes, I have reviewed the [localization checklist](https://github.com/github/docs/blob/main/contributing/localization-checklist.md)
- [x] For content changes, I have reviewed the [Content style guide for GitHub Docs](https://github.com/github/docs/blob/main/contributing/content-style-guide.md).

### CODE REVIEW

**Comment by janiceilene** [comment]
Thanks @nschonni! I'll get this triaged for review ⚡

**Review by heiskr — APPROVED** [review]
👏🏼 This looks like a solid improvement.

**Review by JamesMGreene — CHANGES_REQUESTED** [review]
I love what this change achieves! 😍 

However, I believe this _also_ implies that we will be unable to make the linting job be a **required check** for merging into the `main` branch, which is less desirable. 😕 

I'll confirm with the Actions support folks internally but I'm going to block this PR for now by marking my review as "Request changes". 🛑

**Comment by nschonni** [comment]
No problem, if this way doesn't work, I can try and re-work it with fkirc/skip-duplicate-actions

**Comment by nschonni** [comment]
@JamesMGreene figured out a way to test it by removing the trigger on the config change, and it looks like it will stall with the "Required".

image

I'll try with the other action, and do the same for https://github.com/github/docs/pull/541

**Comment by nschonni** [comment]
@JamesMGreene I can't see if the now "Skipped" required check will block the commit. Can you let me know, and I can in-line the skip check flow into the `lint` job instead

**Review by JamesMGreene — APPROVED** [review]
Approving to check if it meets "Required Checks"... uhh... requirements. 😅

**Comment by nschonni** [comment]
Oops, looks like the branch merge triggered the `lint` check. I'll rebase back to the first commit and re-submit the second one to see if we can get back to the "Skipped" required check

**Comment by nschonni** [comment]
@JamesMGreene OK, it's back to the "Skipped" required check again

File: .github/workflows/js-lint.yml
Comment (JamesMGreene) [inline .github/workflows/js-lint.yml:24]: Let's add some spaces here for legibility. ✨ 

```suggestion
          paths: '["**/*.js", "package*.json", ".github/workflows/js-lint.yml"]'
```

**Review by JamesMGreene — APPROVED** [review]
Good news! ☝🏻  It actually seems to be happy with this. 🎉 

image

That's so awesome, @nschonni. It really cleans the test workflow up nicely. 😍

**Pull request merged**
